### PR TITLE
Updated dependencies for symfony 4 #SymfonyConHackday2017

### PR DESCRIPTION
**Pull request by Jan0707**
With best regards from #SymfonyConHackday2017

### CODE REVIEW

**Comment by sstok** [comment]
You still need to update the composer.json version constraints 😛 otherwise you get the tests working but installing will not be possible.

**Comment by Jan0707** [comment]
@sstok done.

**Comment by DonCallisto** [comment]
@sstok @Jan0707 so it will be safe to require `symfony 4` also we it will be available?

**Comment by Jan0707** [comment]
@DonCallisto That is the idea :)

**Comment by fridde** [comment]
Just saw this PR after checking why no one was able to comment on my [issue](https://github.com/APY/APYDataGridBundle/issues/988).

What's keeping you guys from merging?

File: .travis.yml [hunk 1]
```diff
@@ -12,6 +12,9 @@ matrix:
     - php: 5.6
       env: |
         SYMFONY_VERSION=2.8.*
+    - php: 7.1
+      env: |
+        SYMFONY_VERSION=^4.0-beta
```
Comment (DonCallisto) [inline .travis.yml]: Could you modify this for symfony 4 (not beta?)

File: composer.json [hunk 2]
```diff
@@ -21,14 +21,14 @@
     ],
     "require": {
         "php": ">=5.6",
-        "symfony/symfony": "~2.8|~3.0",
+        "symfony/symfony": "~2.8|~3.0|^4.0-BETA",
```
Comment (DonCallisto) [inline composer.json]: As well

**Comment by DonCallisto** [comment]
@fridde I've made a review. Waiting for @Jan0707 to make this little modification.

File: composer.json (hunk 2 above)
Comment (fridde) [reply composer.json]: Thanks! I really love this project! 
forking it just to add 5 more characters in composer.json would somehow not be the ideal path to go ;-)

Comment (DonCallisto) [reply composer.json]: Of course it isn't but as you can see yourself, in last 12 months or so, I'm the only one who really cares about this project. Sadly I don't have so much free time to dedicate to it

**Comment by Jan0707** [comment]
Sorry folks, I was travelling. I added the requested changes :)

**Comment by DonCallisto** [comment]
I've merged this: what do you guys think about next release? It should be a bug fix version, a minor version (I vote for this) or a new major version? What's semantic versioning suggest here?

@Jan0707 @sstok @fridde

**Comment by Jan0707** [comment]
This introduces no BC breaks, so there is no need for a major revision. However, semantic versioning suggests the following:

> MINOR version when you add functionality in a backwards-compatible manner, and [...]

[See here: https://semver.org](https://semver.org)

**Comment by DonCallisto** [comment]
@Jan0707 I personally agree with this, but I was wondering about https://github.com/Jan0707/APYDataGridBundle/blob/29727dfe7f5930f76ef68acfbf203ac29d12b402/composer.json#L31

You can confirm that semver is still respected even when you do a major update like this?
Sorry but I don't have such experience in third-party releases 😄

**Comment by Jan0707** [comment]
Given that the mentioned dependency is dev only, this should not really matter for releases.

**Comment by DonCallisto** [comment]
[Here we go](https://github.com/APY/APYDataGridBundle/releases/tag/3.1.0)

**Pull request merged**